src/interface/src/app/treatments/direct-impacts/metric-filters/metric-selector/metric-selector.component.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  Component,
+  EventEmitter,
+  Input,
+  OnInit,
+  Output,
+  ViewEncapsulation,
+} from '@angular/core';
+import { MatIconModule } from '@angular/material/icon';
+import { MatSelectModule } from '@angular/material/select';
+import { CommonModule } from '@angular/common';
+import { FormsModule } from '@angular/forms';
+import { Metric } from '../metric-filters.component';
+
+@Component({
+  selector: 'app-metric-selector',
+  standalone: true,
+  imports: [CommonModule, FormsModule, MatIconModule, MatSelectModule],
+  templateUrl: './metric-selector.component.html',
+  styleUrl: './metric-selector.component.scss',
+  encapsulation: ViewEncapsulation.None,
+})
+export class MetricSelectorComponent implements OnInit {
+  @Input() label!: string;
+  @Input() metrics!: Metric[];
+  @Input() selectedOption!: string;
+  @Output() metricSelection = new EventEmitter<Metric>();
```

Comment:
metricSelection is when the user changes the metric on the dropdown, but not when that one is picked as the active one? Maybe `metricChanged` ?

Reply:
Updated, you will see it in the next commit.